backend/src/main/java/com/soen341/instagram/service/impl/AccountService.java

```diff
@@ -0,0 +1,53 @@
+package com.soen341.instagram.service.impl;
+
+import java.util.Date;
+import java.util.regex.Pattern;
+
+// Spring
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Service;
+import com.soen341.instagram.exception.account.InvalidEmailFormatException;
+
+// Project
+import com.soen341.instagram.dao.impl.AccountRepository;
+import com.soen341.instagram.model.Account;
+import com.soen341.instagram.utils.AccountVerifier;
+import com.soen341.instagram.utils.UserAccessor;
+
+@Service
+public class AccountService 
+{
+	@Autowired
+	AccountRepository accountRepository;
+
+  public void setPassword(final String newPassword) {
+      String oldPassword = UserAccessor.getCurrentAccount(accountRepository).getPassword();
+      AccountVerifier.checkIfSamePassword(oldPassword, newPassword);
+      final Account account = UserAccessor.getCurrentAccount(accountRepository);
+      account.setPassword(newPassword);
```

Comment:
The password saved here is not encoded. You could refer to `RegistrationService.createNewAccount()`, it uses a password encoder before saving the password.